sympy/functions/special/zeta_functions.py

```diff
@@ -582,6 +583,44 @@ def _eval_rewrite_as_zeta(self, s, **kwargs):
         return (1 - 2**(1 - s)) * zeta(s)
 
 
+class riemann_xi(Function):
+    r"""
+    Riemann Xi function.
+
+    Examples
+    ========
+
+    The Riemann Xi function is closely related to the Riemann zeta function.
+    The zeros of Riemann Xi function are precisely the non-trivial zeros
+    of the zeta function.
+
+    >>> from sympy import riemann_xi, zeta
+    >>> from sympy.abc import s
+    >>> riemann_xi(s).rewrite(zeta)
+    pi**(-s/2)*s*(s - 1)*gamma(s/2)*zeta(s)/2
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Riemann_Xi_function
+
+    """
+
+
+    @classmethod
+    def eval(cls, s):
+        from sympy import gamma
+        z = zeta(s)
+        if s == 0 or s == 1:
+            return Rational(1,2)
+
+        if not isinstance(z, zeta):
```

Comment:
Why is this check needed?
With this check `riemann_xi(3)` returns `riemann_xi(3)`.
Removing the check will make the results consistent with [wolfram](https://mathworld.wolfram.com/Xi-Function.html)

Reply:
according to wolfram `riemann_xi(0)` or `riemann_xi(1)` is equal to `1/2`. But accroding to the formula `riemann_xi(0)` ealuates 0 multiplied by gamma(0)(`zoo`) which gives `nan` but should be `0`. So to handle such cases for `0` and `1` this check is required.

Reply:
I was asking regarding `if not isinstance(z, zeta):`
`riemann_xi(3)` is in terms of zeta functions only as mentioned in the link above, and with this condition that would be missed so only `riemann_xi(3)` would be output, and not the answer given on wolfram.

Reply:
`riemann_xi(3)` has a `zeta(3)` in it so if we do `riemann_xi(3).rewrite(zeta)` it would give the desired output. This is for the riemann x which are in terms of zeta functions.